internal/rootcoord/meta_snapshot.go

```diff
@@ -104,9 +105,18 @@ func (ms *metaSnapshot) loadTs() error {
 		curVer := resp.Kvs[0].Version
 		curRev := resp.Kvs[0].ModRevision
 		if curVer > version {
+			log.Warn("version go backwards", zap.Int64("curVer", curVer), zap.Int64("version", version))
 			return nil
 		}
+		if curVer == version {
+			log.Debug("snapshot found save version with different revision", zap.Int64("revision", revision), zap.Int64("version", version))
+		}
 		strTs := string(resp.Kvs[0].Value)
+		if strTs == "0" {
+			//#issue 7150, index building inserted "0", skipping
+			//this is a special fix for backward compatibility, previous version will put 0 ts into snapshot building index
+			continue
+		}
```

Comment:
This part filters “0” value ts from older version. It's just for backward compatibility. Shall we keep this snippet only for handling data error in RC period? @sunby @xiaofan-luan @czs007 @scsven

Reply:
I suggest keeping this code, but adding some conversion logic to change 0ts to non-0ts storage. Delete this code when the GA version is released.

Reply:
> I suggest keeping this code, but adding some conversion logic to change 0ts to non-0ts storage. Delete this code when the GA version is released.

This suggestion looks good to me

Reply:
> I suggest keeping this code, but adding some conversion logic to change 0ts to non-0ts storage. Delete this code when the GA version is released.

I don't think there is a way to modify etcd kv with specified revision. Maybe we needs to change the implementation of `SnapshotKV`.